### PR TITLE
[CI-CD Refactor] Complete the missing features before tackling validation ticket.

### PR DESCRIPTION
**Pull request by dbogunowicz**
Addressing several tiny todos:
- @rahul-tuli comments from the previous (merged) PR
- contains all the changes introduced by #161 (but I will close the old PR, the state of `feature/ci-cd-refactor/main` has changed too much in the meantime)
- integrate the final model structure as discussed with @markurtz  and @bfineran

### CODE REVIEW

File: src/sparsezoo/v2/model_directory.py [hunk 1]
```diff
@@ -449,6 +466,40 @@ def _download(
...
+        if not isinstance(directories, list):
+            # if found a single 'sample_outputs' directory,
+            # assume it should be inferred using the native framework
+            expected_name = "sample_outputs"
+            if directories.name != expected_name:
+                raise ValueError(
+                    "Found single folder (or tar.gz archive)"
+                    "with expected name `sample_outputs`. However,"
+                    f"detected a name {directories.name}"
+                    f"It's name should be {expected_name}"
```
Comment (dbogunowicz) [inline src/sparsezoo/v2/model_directory.py:483]:
```suggestion
                    f"It's name should be {expected_name}."
```

File: src/sparsezoo/v2/model_directory.py [hunk 2]
```diff
@@ -258,7 +272,7 @@ def validate(self) -> bool:
                 for _file in file:
                     validations[_file.name] = _file.validate()
             elif isinstance(file, dict):
-                raise NotImplementedError()
+                pass
```
Comment (dbogunowicz) [inline src/sparsezoo/v2/model_directory.py:275]: This will be addressed in the next PR.

File: src/sparsezoo/v2/inference_runner.py [hunk 3]
```diff
@@ -88,59 +91,56 @@ def generate_outputs(
...
-                output_files.append(File(name=name, path=os.path.join(path, name)))
-
-            output_directory = NumpyDirectory(
-                name=os.path.basename(path), path=path, files=output_files
-            )
-            output_directory.gzip()
+            self._save_outputs_to_tar(iterator, engine_type)
 
-        for output in outputs:
+        for output in iterator():
```
Comment (rahul-tuli) [inline src/sparsezoo/v2/inference_runner.py:97]: nice!

**Review by rahul-tuli — APPROVED** [review]

File: src/sparsezoo/v2/model_directory.py [hunk 4]
```diff
@@ -183,7 +197,7 @@ def generate_outputs(
             by the `inference_runner` will be additionally saved to
             the archive file `sample_outputs_{engine_type}.tar.gz
             (located in the `self.path` directory).
-        :returns returns a data structure
+        :returns returns list
```
Comment (KSGulin) [inline src/sparsezoo/v2/model_directory.py:200]:
```suggestion
        :returns list
```

**Review by KSGulin — COMMENTED** [review]
Looks great. Only had a couple nits

**Review by bfineran — APPROVED** [review]

**Pull request merged**